libraries/AP_Parachute/AP_Parachute.cpp

```diff
@@ -86,6 +86,12 @@ void AP_Parachute::enabled(bool on_off)
     // clear release_time
     _release_time = 0;
 
+    // Matternet manufacturing test: Arm FTS on parachute enable
+    if (on_off) {
+        hal.util->set_soft_armed(true);
```

Comment:
Does the AP stay armed long enough for parachute release or does it disarm after the arm timeout?

Reply:
It stays Armed indefinitely.

Reply:
Here is a printout of the message status to the FTS. Each line is 1 second interval:

% mavproxy.py --master /dev/ttyACM0

AP: FTS: Disarm
AP: FTS: Disarm
AP: FTS: Disarm
AP: FTS: Disarm
AP: FTS: Disarm
AP: FTS: Disarm
AP: FTS: Disarm
AP: FTS: Disarm

STABILIZE> parachute enable
STABILIZE> AP: Parachute: Enabled
Got COMMAND_ACK: DO_PARACHUTE: ACCEPTED
AP: FTS: Arm
AP: FTS: Arm
AP: FTS: Arm
AP: FTS: Arm
AP: FTS: Arm
AP: FTS: Arm
AP: FTS: Arm
AP: FTS: Arm
AP: FTS: Arm
AP: FTS: Arm

STABILIZE> parachute release
STABILIZE> AP: Parachute: Released
Got COMMAND_ACK: DO_PARACHUTE: ACCEPTED
AP: FTS: Deploy
AP: FTS: Deploy
AP: FTS: Deploy
AP: FTS: Deploy
AP: FTS: Deploy
AP: FTS: Deploy
AP: FTS: Deploy

Reply:
Does it stay armed beyond 20 seconds? That's the default config in our build: DISARM_DELAY	20

Reply:
Though if the test script calls enable then deploy, then it doesn't matter if it's beyond 20s. I was concerned that the parachute would be enabled sooner and the timeout would expire before the deploy test script would be executed

Reply:
I'll check that

Reply:
It seems to stay armed indefinitely

Reply:
There is "soft_armed" and "armed".  What is the difference between the two? I am controlling soft_armed because the FTS logic checks that variable:

```
      if (!_release_initiated) {
            if (hal.util->get_soft_armed()) {
                struct fts_msg_arm_cmd_s arm_msg;
                arm_msg.msgid = FTS_MSGID_ARM_CMD;
                mttr_fts_transmit(sizeof(arm_msg), (uint8_t*)&arm_msg);
                gcs().send_text(MAV_SEVERITY_INFO,"FTS: Arm");
            } else {
                struct fts_msg_disarm_cmd_s disarm_msg;
                disarm_msg.msgid = FTS_MSGID_DISARM_CMD;
                mttr_fts_transmit(sizeof(disarm_msg), (uint8_t*)&disarm_msg);
                gcs().send_text(MAV_SEVERITY_INFO,"FTS: Disarm");
            }
        }
```